domain/src/main/java/org/oppia/android/domain/exploration/lightweightcheckpointing/ExplorationCheckpointController.kt

```diff
@@ -280,4 +264,22 @@ class ExplorationCheckpointController @Inject constructor(
 
   @VisibleForTesting(otherwise = VisibleForTesting.NONE)
   fun getExplorationCheckpointDatabaseSizeLimit(): Int = explorationCheckpointDatabaseSizeLimit
+
+  @VisibleForTesting(otherwise = VisibleForTesting.NONE)
```

Comment:
We ought to never use VisibleForTesting. Can we test this through the methods that call it, instead? Ie, that verifies the expected production behavior: under some conditions, those public API functions should result in a checkpoint being saved.

Reply:
For this PR, this function is being used to test if checkpoints are being saved, since the function to save checkpoints in the production code returns a deferred so for testing, this function creates a dataProvider from that deferred.

For the next PR this function will be used to save checkpoints so that we can have multiple checkpoints saved in the database to properly test those features of checkpointing that are not directly related with saving checkpoints.

If we remove this function, we can create `dataProviders` from the deferred in the classes `ExplorationCheckpointControllerTest` and `ExplorationCheckpointTestHelper` (added in the next PR).

@BenHenning WDYT?

Reply:
Either this function is part of the class's API (which means it should be public/internal & have a KDoc explaining out to use it) in which case it'll have dedicated tests, or it's a helper for other functions in this controller & should be tested indirectly through those. In neither case should VIsibleForTesting be needed.

It seems like the latter since you have the async version of this function above. Could we use that to test this, instead?

Reply:
removed `visible for testing`. Now, this function is part of the class's API and is being used in testing.